src/chessboard/index.tsx

```diff
@@ -25,53 +25,92 @@ export type ClearPremoves = {
   clearPremoves: (clearLastPieceColour?: boolean) => void;
 };
 
-export const Chessboard = forwardRef<ClearPremoves, ChessboardProps>((props, ref) => {
-  const { customDndBackend, customDndBackendOptions, ...otherProps } = props;
-  const [clientWindow, setClientWindow] = useState<Window>();
-  const [backendSet, setBackendSet] = useState(false);
-  const [isMobile, setIsMobile] = useState(false);
-  const [boardWidth, setBoardWidth] = useState(props.boardWidth);
+export const Chessboard = forwardRef<ClearPremoves, ChessboardProps>(
+  (props, ref) => {
+    const { customDndBackend, customDndBackendOptions, ...otherProps } = props;
+    const [clientWindow, setClientWindow] = useState<Window>();
+    const [backendSet, setBackendSet] = useState(false);
+    const [isMobile, setIsMobile] = useState(false);
+    const [boardWidth, setBoardWidth] = useState(props.boardWidth);
 
-  const boardRef = useRef<HTMLObjectElement>(null);
+    const boardRef = useRef<HTMLObjectElement>(null);
+    const boardContainerRef = useRef<HTMLDivElement>(null);
 
-  useEffect(() => {
-    setIsMobile("ontouchstart" in window);
-    setBackendSet(true);
-    setClientWindow(window);
-  }, []);
+    const [boardContainerPos, setBoardContainerPos] = useState({
+      left: 0,
+      top: 0,
+    });
 
-  useEffect(() => {
-    if (props.boardWidth === undefined && boardRef.current?.offsetWidth) {
-      const resizeObserver = new ResizeObserver(() => {
-        setBoardWidth(boardRef.current?.offsetWidth as number);
+    //Set BoardContainerPos initially when boardContainerRef is set. Position is used when dragOutsideOfBoard={false}
+    useEffect(() => {
+      if (boardContainerRef.current) {
+        const { left, top } = boardContainerRef.current.getBoundingClientRect();
+        setBoardContainerPos({ left, top });
+      }
+    }, [boardContainerRef.current]);
+
+    useEffect(() => {
+      window.addEventListener("resize", (e) => {
+        if (boardContainerRef.current) {
+          const { left, top } =
+            boardContainerRef.current.getBoundingClientRect();
+          setBoardContainerPos({ left, top });
+        }
+        return () => window.removeEventListener("resize", (e) => {});
       });
-      resizeObserver.observe(boardRef.current);
+    }, []);
+
+    useEffect(() => {
+      setIsMobile("ontouchstart" in window);
+      setBackendSet(true);
+      setClientWindow(window);
+    }, []);
+
+    useEffect(() => {
+      if (props.boardWidth === undefined && boardRef.current?.offsetWidth) {
+        const resizeObserver = new ResizeObserver(() => {
+          setBoardWidth(boardRef.current?.offsetWidth as number);
```

Comment:
```suggestion
          setBoardWidth(boardRef.current?.offsetWidth as number);
          // try this instead
          const metrics = boardRef.current?.getBoundingClientRect();
          console.log(">>>>>>>>", {
            offestTop: boardRef.current?.offsetTop,
            offestLeft: boardRef.current?.offsetLeft,
            clientRectTop: metrics?.top,
            clientRectLeft: metrics?.left,
          });
```

Reply:
I have updated it to use that but still had to wrap it in a useEffect to prevent too many rerenders.